### PR TITLE
Laboratory work #7, Maria Styagova - 20FPL2

### PR DESCRIPTION
**Pull request by styagovamaria**

### CODE REVIEW

File: lab_7_llm/main.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
     Dataset = dict
     torch = namedtuple('torch', 'no_grad')(lambda: lambda fn: fn)  # type: ignore
 
+
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py]: Delete line

**Comment by SalnikovIgor** [comment]
Necessary implementation of loading your dataset.

File: lab_7_llm/main.py [hunk 2]
```diff
@@ -41,6 +46,7 @@ def obtain(self) -> None:
         Raises:
             TypeError: In case of downloaded dataset is not pd.DataFrame
         """
+        self._raw_data = DataFrame(load_dataset(self._hf_name, split='if_test'))
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py]: Use the method `to_pandas` after calling `load_dataset`, as such an implementation will be cleaner.

**Comment by SalnikovIgor** [comment]
Everything is fine 👍 Time to start the task at 6

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]
Time to start the task at 6

File: lab_7_llm/main.py [hunk 3]
```diff
@@ -55,12 +60,25 @@ def analyze(self) -> dict:
...
         """
+        rows, cols = self._raw_data.shape
+        data_droped_empty = self._raw_data.dropna()
+
+        return {'dataset_number_of_samples': rows,
+                'dataset_columns': cols,
+                'dataset_duplicates': len(self._raw_data[self._raw_data.duplicated()]),
+                'dataset_empty_rows': rows - len(data_droped_empty),
+                'dataset_sample_min_len': min(data_droped_empty['ru'].str.len()),
+                'dataset_sample_max_len': max(data_droped_empty['ru'].str.len())}
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py:68]: Use `ColumnNames.SOURCE.value` instead of `ru`

File: lab_7_llm/main.py [hunk 4]
```diff
@@ -94,6 +114,7 @@ def __getitem__(self, index: int) -> tuple[str, ...]:
         Returns:
             tuple[str, ...]: The item to be received
         """
+        return self._data.iloc[index][ColumnNames.SOURCE], self._data.iloc[index][ColumnNames.TARGET]
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py]: It is necessary to return only the column `ColumnNames.SOURCE`

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]

File: lab_7_llm/main.py (hunk 3 above)
Comment (styagovamaria) [reply lab_7_llm/main.py:68]: I don't understand. This method is called before ```transform```, so the column name is ```ru``` not ```source```. Or should I use my own enum for this?

Comment (SalnikovIgor) [reply lab_7_llm/main.py:68]: I forgot that we are renaming the columns. Then your decision is correct

File: lab_7_llm/start.py [hunk 5]
```diff
@@ -2,15 +2,45 @@
...
+    assert importer.raw_data is not None, "Demo does not work correctly"
+    preprocessor = RawDataPreprocessor(importer.raw_data)
+    print(preprocessor.analyze())
+    preprocessor.transform()
+
+    dataset = TaskDataset(preprocessor.data.head(100))
+    pipeline = LLMPipeline(settings['parameters']['model'], dataset, 120, 64, 'cpu')
+    print(pipeline.analyze_model())
+    infer_sample_result = pipeline.infer_sample(dataset[0])
+    print(f'INPUT TEXT: {dataset[0][0]}\nTRANSLATION: {infer_sample_result}')
```
Comment (SalnikovIgor) [inline lab_7_llm/start.py]: If this is debugging information, then delete it

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]

File: lab_7_llm/start.py (hunk 5 above)
Comment (styagovamaria) [reply lab_7_llm/start.py]: This is a part of task for 6 with ```infer_sample``` work demonstration but I removed it.

**Comment by styagovamaria** [comment]
image
Oh, the tests failed because the HuggingFace is not available.

**Comment by SalnikovIgor** [comment]
> image Oh, the tests failed because the HuggingFace is not available.

Push changes, when the HuggingFace comes to life

**Comment by styagovamaria** [comment]
> > image Oh, the tests failed because the HuggingFace is not available.
> 
> Push changes, when the HuggingFace comes to life

Done, it's ok now

**Pull request closed without merging**